core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -205,18 +205,18 @@ public void commit() throws IOException {
             try {
                 Files.move(tmpPath, destPath, StandardCopyOption.REPLACE_EXISTING);
             } catch (IOException e1) {
-                e1.addSuppressed(e);
+                e.addSuppressed(e1);
                 LOGGER.log(Level.WARNING, "Unable to move {0} to {1}. Attempting to delete {0} and abandoning.",
                            new Path[]{tmpPath, destPath});
                 try {
                     Files.deleteIfExists(tmpPath);
                 } catch (IOException e2) {
-                    e2.addSuppressed(e1);
+                    e1.addSuppressed(e2);
```

Comment:
If my previous suggestion is accepted, then I think we would want to add e2 to e as the suppressed exception so that the very first relevant exception is thrown, whether it is the original non-AtomicMoveNotSupportedException exception or the e1 exception after an AtomicMoveNotSupportedException.
```suggestion
                    e.addSuppressed(e2);
```